### DISCUSSION TITLE
Add a PIN for Bluetooth

### DISCUSSION THREAD

**Discussion by MrGibbage** [pybricks/support#1904] [Ideas]
Could we add an optional PIN for BT connections? I have heard people say in different places that their robots were taken over by other teams (hopefully not maliciously, but who knows). Anyway, I would suppose the PIN could be set when flashing the firmware, and then it could be an optional parameter for pybricksdev. So many teams have their robot names written on them that it would be quite easy to make a connection to them right now. It wouldn't be super-secure, but maybe just enough secure to prevent problems. Especially as pybricks gains in popularity, this might be very useful. 

For the record, I have not personally had this problem, and I am pretty sure the people who did have problems were using Lego firmware, but maybe a good time for pybricks to get in front of this before it becomes a problem.

**Comment by dlech** [comment]
Just to keep ideas tied together, @afarago also mentioned something like this this in https://github.com/pybricks/support/issues/1716#issuecomment-2308293859.

**Comment by laurensvalk** [comment]
In the latest beta version you can turn Bluetooth off by pressing the Bluetooth button on the hub, which should also help make take overs (accidental or otherwise) less likely.

**Comment by MrGibbage** [comment]
LOL, what's actually happening more often now is the kids accidentally turn BT off and we spend too much time trying to figure out why all of a sudden we can't upload our program. It's happened enough times now that I tend to think about it, but boy the first time it happened, I probably scratched my head for a good hour after practice trying to fix the robot/code.

That #1716 thread is really interesting. Some great ideas in there.

Btw, here's the error you get when the laptop can't find the robot:

````
Executing task: C:\Users\Skip\GithubProjects\FLL-Fall-2024-Submerged/.venv/Scripts/pybricksdev.exe run ble --name PATRIOT C:\Users\Skip\GithubProjects\FLL-Fall-2024-Submerged\master_program.py

Searching for PATRIOT...
Traceback (most recent call last):
  File "C:\Users\Skip\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Skip\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\Skip\GithubProjects\FLL-Fall-2024-Submerged\.venv\Scripts\pybricksdev.exe\__main__.py", line 7, in <module>
  File "c:\Users\Skip\GithubProjects\FLL-Fall-2024-Submerged\.venv\lib\site-packages\pybricksdev\cli\__init__.py", line 380, in main
    asyncio.run(subparsers.choices[args.tool].tool.run(args))
  File "C:\Users\Skip\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\Skip\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 646, in run_until_complete
    return future.result()
  File "c:\Users\Skip\GithubProjects\FLL-Fall-2024-Submerged\.venv\lib\site-packages\pybricksdev\cli\__init__.py", line 192, in run
    device_or_address = await find_device([args.name](http://args.name/))
  File "c:\Users\Skip\GithubProjects\FLL-Fall-2024-Submerged\.venv\lib\site-packages\pybricksdev\ble\__init__.py", line 68, in find_device
    raise asyncio.TimeoutError
asyncio.exceptions.TimeoutError

 *  The terminal process "C:\WINDOWS\System32\cmd.exe /d /c C:\Users\Skip\GithubProjects\FLL-Fall-2024-Submerged/.venv/Scripts/pybricksdev.exe run ble --name PATRIOT C:\Users\Skip\GithubProjects\FLL-Fall-2024-Submerged\master_program.py" terminated with exit code: 1.

````

Slight tangent here, but is there any way this error could also give a big kid-friendly hint to check the bluetooth on the robot? The current error message is too long, and the kids see it and think something terrible must have happened. Something like:
"Sorry, but I couldn't find PATRIOT. Check to see if the bluetooth is turned on, and make sure you spelled the robot name correctly."
